### PR TITLE
Add document for Scheme manifest

### PR DESCRIPTION
**Pull request by dangthaison91**
Addition document for Scheme

### CODE REVIEW

File: docs/usage/2-manifest.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ name: Manifest format
 Projects are defined in `Project.swift` files, which we refer to as manifest files. Manifests need to import the framework `ProjectDescription` which contains all the classes and enums that are available for you to describe your projects. The snippet below shows an example project manifest:
 
 ```swift
+let releaseScheme = Scheme(name: "App-Release",
```
Comment (kwridan) [inline docs/usage/2-manifest.md]: Looks like this was added by mistake, the project manifest has an inline scheme

File: docs/usage/2-manifest.md [hunk 2]
```diff
@@ -37,6 +55,7 @@ A `Project.swift` should initialize a variable of type `Project`. It can take an
 
 - **Name:** Name of the project. It's used to determine the name of the generated Xcode project.
 - **Targets:** It contains the list of targets that belong to the project. Read more [about targets](#target)
+- **Schemes:** It contains the list of schemes will be generated . Read more [about schemes](#scheme)
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
- **Schemes:** It contains the list of schemes that will be generated . Read more [about schemes](#scheme)
```

File: docs/usage/2-manifest.md [hunk 3]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
   - **CurrentVersion:** Current version without the extension.
 - **Environment (optional):** It's a `[String: String]` to defined variables that will be set to the scheme that Tuist automatically generates for the target.
 
+### Scheme
+
+A `Scheme` will help to define a collection of targets to `Build, Run, Test, Profile, Analyze and Archive`:
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
A `Scheme` defines a collection of targets to `Build, Run, Test, Profile, Analyze and Archive`:
```

File: docs/usage/2-manifest.md [hunk 4]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+> Note: Tuist will auto-generate Scheme for each target if they are not defined.
```
Comment (kwridan) [inline docs/usage/2-manifest.md]: 👍

File: docs/usage/2-manifest.md [hunk 5]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
 - **Environment (optional):** It's a `[String: String]` to defined variables that will be set to the scheme that Tuist automatically generates for the target.
 
+### Scheme
+
+A `Scheme` will help to define a collection of targets to `Build, Run, Test, Profile, Analyze and Archive`:
+> Note: Tuist will auto-generate Scheme for each target if they are not defined. 
+- **name: String:** Name of Scheme
+- **shared: Bool:** Scheme is shared or not 
+- **buildAction: BuildAction?:** Scheme Build options
+    - **targets: [String]:** A list of targets will be built, which are defined in Project
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
    - **targets: [String]:** A list of targets to build, which are defined in the Project
```

Comment (kwridan) [reply docs/usage/2-manifest.md]: This actually surfaces a future enhancement - technically Xcode does all you to specify any target in the workspace not just the project 🙂

File: docs/usage/2-manifest.md [hunk 6]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+    - **preActions:[ [ExecutionAction](#execution-action)]:** A list of pre build actions
+    - **postActions: [[ExecutionAction](#execution-action)]:** A list of post build actions
+- **testAction: TestAction?:** Scheme Test options
+    - **targets: [String]:** A list of targets will be tested, which are defined in Project
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
    - **targets: [String]:** A list of targets to test, which are defined in the Project
```

File: docs/usage/2-manifest.md [hunk 7]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+    - **arguments: [Arguments?](#arguments):** Commandline arguments passed on launch and Environment variables
+    - **config: BuildConfiguration:** Test with  `debug` or `release` configuration
+    - **coverage: Bool:** Scheme should gather coverage data or not. Default is `false`
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
    - **coverage: Bool:** Enables code coverage. Default is `false`
```

Good idea documenting the defaults too 👍

File: docs/usage/2-manifest.md [hunk 8]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
   - **CurrentVersion:** Current version without the extension.
 - **Environment (optional):** It's a `[String: String]` to defined variables that will be set to the scheme that Tuist automatically generates for the target.
 
+### Scheme
+
+A `Scheme` will help to define a collection of targets to `Build, Run, Test, Profile, Analyze and Archive`:
+> Note: Tuist will auto-generate Scheme for each target if they are not defined. 
+- **name: String:** Name of Scheme
+- **shared: Bool:** Scheme is shared or not
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
- **shared: Bool:** Marks the scheme as shared (i.e. one that is checked in to the repository and is visible to `xcodebuild` from the command line)
```

File: docs/usage/2-manifest.md [hunk 9]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+- **testAction: TestAction?:** Scheme Test options
+    - **targets: [String]:** A list of targets will be tested, which are defined in Project
+    - **arguments: [Arguments?](#arguments):** Commandline arguments passed on launch and Environment variables
+    - **config: BuildConfiguration:** Test with  `debug` or `release` configuration
+    - **coverage: Bool:** Scheme should gather coverage data or not. Default is `false`
+    - **preActions: [[ExecutionAction](#execution-action)]:** Pre test script action
+    - **postActions: [[ExecutionAction](#execution-action)]:** Post test script action
+- **runAction: RunAction?:** Run or Launch action
+    - **config: BuildConfiguration:** Run with `debug` or `release` configuration
+    - **executable: String?:** Executable or target will be launch
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
    - **executable: String?:** Executable or target to launch
```

File: docs/usage/2-manifest.md [hunk 10]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+    - **arguments: [Arguments?](#arguments):** Commandline arguments passed on launch and Environment variables
+    
+#### Execution Action
+Scheme run scripts can be defined with the following attributes:
+- **title: String:** Name of a script. Defaults to `Run Script`
+- **scriptText: String:** An inline shell script
+- **target: String?:** Name of a build or test target that will provide build settings.
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
- **target: String?:** Name of the build or test target that will provide the action's build settings.
```

File: docs/usage/2-manifest.md [hunk 11]
```diff
@@ -88,6 +107,39 @@ Each target in the list of project targets can be initialized with the following
...
+
+#### Arguments
+Arguments contain commandline arguments passed on launch and Environment variables
+- **environment: [String: String]:** Environment variables
+- **launch: [String: Bool]:** Commandline arguments
```
Comment (kwridan) [inline docs/usage/2-manifest.md]:
```suggestion
- **launch: [String: Bool]:** Commandline launch arguments where the key is a launch argument and the value is a boolean indicating if it's enabled
```

File: docs/usage/2-manifest.md (hunk 10 above)
Comment (pepibumur) [inline docs/usage/2-manifest.md]: In which scenarios can this attribute be optional? I'd give it a default value, `nil`.

File: docs/usage/2-manifest.md (hunk 1 above)
Comment (dangthaison91) [reply docs/usage/2-manifest.md]: I intended to extract Scheme manifest into separate variable for better readability but later have changed it to inline to keep document consistent.

File: docs/usage/2-manifest.md (hunk 5 above)
Comment (dangthaison91) [reply docs/usage/2-manifest.md]: Yes, at the moment we cannot generate Scheme with targets from different Project as Xcode does.

**Pull request merged**